src/Microsoft.HttpRepl/HttpState.cs

```diff
@@ -83,10 +85,15 @@ public IEnumerable<string> GetApplicableContentTypes(string method, string path)
 
         public Uri GetEffectivePath(string commandSpecifiedPath)
         {
-            return GetEffectivePath(BaseAddress, string.Join('/', PathSections.Reverse()), commandSpecifiedPath);
+            return GetEffectivePath(BaseAddress, string.Join('/', PathSections.Reverse()), commandSpecifiedPath, queryParam: null);
+        }
+
+        public Uri GetEffectivePathWithQueryParam(string commandSpecifiedPath, Dictionary<string, IEnumerable<string>> queryParam)
```

Comment:
A few things seem wrong here:

1. We have GetEffectivePathWithQueryParam that takes the queryParam parameter when the only call into it uses the query params that are stored in this object.
2. We have three public GetEffectivePath* methods, even though only one is used outside of this class (or test cases).
3. The one GetEffectivePath* method that is used outside of this class is the lengthier-named one.

So I have a suggestion. What if we instead had (psuedo-code):

```C#
private static Uri GetEffectivePath(Uri baseAddress, string pathSections, string commandSpecifiedPath, Dictionary<string, IEnumerable<string>> queryParam)
{
    // The same as the current static method
}

private Uri GetEffectivePathWithoutQueryParam(string commandSpecified)
    => GetEffectivePath(BaseAddress, string.Join('/', PathSections.Reverse()), commandSpecifiedPath, queryParam: null);

public Uri GetEffectivePath(string commandSpecified)
    => GetEffectivePath(BaseAddress, string.Join('/', PathSections.Reverse()), commandSpecifiedPath, QueryParam);
```

(private may need to be internal for tests)

And then change the current calls to GetEffectivePath and GetEffectivePathWithQueryParam to the appropriate new one to match the current behavior (or adjusted behavior per other comments).

Thoughts?

Reply:
Sounds good to me, the latest commit addresses these changes!